samtranslator/swagger/swagger.py

```diff
@@ -131,12 +132,25 @@ def add_disable_execute_api_endpoint_extension(self, disable_execute_api_endpoin
         https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-endpoint-configuration.html
         :param boolean disable_execute_api_endpoint: Specifies whether clients can invoke your API by using the default execute-api endpoint.
         """
-        if not self._doc.get(self._X_ENDPOINT_CONFIG):
-            self._doc[self._X_ENDPOINT_CONFIG] = {}
-
         DISABLE_EXECUTE_API_ENDPOINT = "disableExecuteApiEndpoint"
         set_disable_api_endpoint = {DISABLE_EXECUTE_API_ENDPOINT: disable_execute_api_endpoint}
-        self._doc[self._X_ENDPOINT_CONFIG].update(set_disable_api_endpoint)
+
+        # Check if the OpenAPI version is 3.0
+        if self._doc.get("openapi") and SwaggerEditor.safe_compare_regex_with_string(  # type: ignore[no-untyped-call]
+            SwaggerEditor.get_openapi_version_3_regex(), self._doc["openapi"]  # type: ignore[no-untyped-call]
```

Comment:
Since this is copied from elsewhere, a static method abstracting it would be much clearer.

Reply:
added static method